asn1/InterledgerProtocol.asn

```diff
@@ -4,76 +4,52 @@ AUTOMATIC TAGS ::=
 BEGIN
 
 IMPORTS
-    UInt64
+    UInt64,
+    UInt256
     FROM GenericTypes
 
     Address,
     Timestamp
     FROM InterledgerTypes
 ;
 
--- the following packet type is experimental and may still change
-ForwardedPaymentData ::= SEQUENCE {
-    -- Destination ILP Address
-    account Address,
-    -- Information for recipient (transport layer information)
-    data OCTET STRING (SIZE (0..32767)),
-    -- Enable ASN.1 Extensibility
-    extensions SEQUENCE {
-        ...
-    }
-}
-
-InterledgerProtocolPaymentData ::= SEQUENCE {
-    -- Amount which must be received at the destination
+InterledgerPrepare ::= SEQUENCE {
+    -- Local amount (changes at each hop)
     amount UInt64,
+
+    -- Execution condition
+    executionCondition UInt256,
+
+    -- Expiry date
+    expiresAt Timestamp,
+
     -- Destination ILP Address
-    account Address,
+    destination Address,
+
     -- Information for recipient (transport layer information)
-    data OCTET STRING (SIZE (0..32767)),
-    -- Enable ASN.1 Extensibility
-    extensions SEQUENCE {
-        ...
-    }
+    data OCTET STRING (SIZE (0..32767))
 }
 
-InterledgerProtocolErrorData ::= SEQUENCE {
-    -- Standardized error code
-    code IA5String (SIZE (3)),
-    -- Corresponding error name
-    name IA5String,
-    -- Participant that originally emitted the error
-    triggeredBy Address,
-    -- Participants that forwarded the error
-    forwardedBy SEQUENCE OF Address,
-    -- Time of emission
-    triggeredAt Timestamp,
-    -- Additional data
-    data OCTET STRING (SIZE (0..8192)),
-    -- Enable ASN.1 Extensibility
-    extensions SEQUENCE {
-        ...
-    }
-}
+InterledgerFulfill ::= SEQUENCE {
+    -- Execution condition fulfillment
+    fulfillment UInt256,
 
-InterledgerProtocolFulfillmentData ::= SEQUENCE {
     -- Information for sender (transport layer information)
     data OCTET STRING (SIZE (0..32767))
 }
 
-InterledgerProtocolRejectionData ::= SEQUENCE {
+InterledgerReject ::= SEQUENCE {
```

Comment:
I'm probably just not understanding the totality of the new flow, but why is the `forwardedBy` no longer in this error?

Reply:
The flow is the same but we thought the `forwardedBy` field was unnecessary. Connectors can pass the error back without needing to modify it to add their address. The feeling was that this field was a bit annoying to deal with, it has the potential to get very large and be abused, and doesn't provide that much value because you mostly care about knowing who threw the error.